File: js/addNum.js
const addNum = {
  num: 3,
  to: function (a) {a.unshift(this.num); return a},
  setNum: function (n) {this.num = n}
}

const a = [1, 2];

console.log('a is ', a);
console.log('addNum output', addNum.to(a));
console.log('a is ', a);

File: java/vavrfun/src/main/java/AddNum.java
import java.util.List;

public class AddNum {
    Integer num;
    AddNum(Integer n) { num = n; }
    public List<Integer> to(List<Integer> ls) {
        ls.add(0, num);
        return ls;
    }
    public void setNum(Integer n) {num = n;}
}
